main.cc

```diff
@@ -1629,6 +1629,8 @@ sharded<locator::shared_token_metadata> token_metadata;
                 api::unset_server_gossip(ctx).get();
             });
 
+            checkpoint(stop_signal, "gossiper API registered");
```

Comment:
I think this `checkpoint` function is not supposed to be used this way. It should be used before starting a service and we have
```cpp
checkpoint(stop_signal, "starting direct failure detector pinger service");
```
below. The call you added is only causing a new log. You can directly use `startlog` instead.

A small issue is that this log is not useful outside tests. It should be a debug log. We don't have any debug/trace logs in main, which makes me a bit uncertain, but it should be fine.

Another option is to wait for some other log between starting gossiper API and joining group 0, but it would be less clear and vulnerable to changes.